Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectILLReductionDIFF.py

```diff
@@ -0,0 +1,147 @@
+        self.output = self.getPropertyValue('OutputWorkspace')
+        self.progress = Progress(self, start=0.0, end=1.0, nreports=10)
+
+    def PyInit(self):
+        self.declareProperty(MultipleFileProperty('SampleRuns', extensions=['nxs']), doc="File path for run(s).")
+
+        self.declareProperty(MatrixWorkspaceProperty('OutputWorkspace', '', direction=Direction.Output),
+                             doc='The output workspace group containing reduced data.')
+
+        self.declareProperty("MaskPixelsFromStart", 10, validator=IntBoundedValidator(lower=0),
+                             doc="Number of pixels to mask at the start of each tube")
+        self.declareProperty("MaskPixelsFromEnd", 10, validator=IntBoundedValidator(lower=0),
+                             doc="Number of pixels to mask at the end of each tube")
+
+        self.declareProperty("Observable", "sample.temperature",
+                             doc="If multiple files, the parameter from SampleLog to use as an index when conjoined.")
+
+    def _normalize_by_monitor(self, ws):
+        """
+            Normalizes the workspace by monitor values (ID is 0 for IN16B)
+            @param ws : the input workspace
+        """
+        monitor_ws = ws + '_mon'
+        ExtractMonitors(InputWorkspace=ws, DetectorWorkspace=ws, MonitorWorkspace=monitor_ws)
+
+        # in case of 0 counts monitors, replace 0s by 1s so the division becomes neutral
+        # (since it's generally division of 0 detector's counts by 0 monitor's counts,
+        # they weren't very useful to begin with)
+
+        ReplaceSpecialValues(InputWorkspace=monitor_ws, OutputWorkspace=monitor_ws, SmallNumberThreshold=0.00000001,
+                             SmallNumberValue=1)
+
+        Divide(LHSWorkspace=ws, RHSWorkspace=monitor_ws, OutputWorkspace=ws, WarnOnZeroDivide=True)
+
+        cache = list(range(1, self.mask_start_pixels)) + list(range(257 - self.mask_end_pixels, 257))
+        to_mask = [i + 256 * j for i in cache for j in range(8)]
+        MaskDetectors(Workspace=ws, DetectorList=to_mask)
+        DeleteWorkspace(monitor_ws)
+
+    def _treat_doppler(self, ws):
+        """
+            Reduce Doppler diffraction data presents in workspace.
+            @param ws: the input workspace
+        """
+        run = None
+        if len(self.runs) > 1:
+            number_of_channels = mtd[mtd[ws].getNames()[0]].blocksize()
+            run = mtd[mtd[ws].getNames()[0]].getRun()
+        else:
+            number_of_channels = mtd[ws].blocksize()
+            run = mtd[ws].getRun()
+
+        if run.hasProperty('Doppler.incident_energy'):
+            energy = run.getLogData('Doppler.incident_energy').value / 1000
+        else:
+            raise RuntimeError("Unable to find incident energy for Doppler mode")
+
+        Rebin(InputWorkspace=ws, OutputWorkspace=self.output, Params=[0, number_of_channels, number_of_channels])
```

Comment:
Is this where you are integrating over all the channels? Can you use Integration instead? For monitor, we should be normalizing to its sum and not bin-to-bin.

Reply:
It's not bin-to-bin, but still detector-to-detector, correct ?

Reply:
Wait, it's after integration, so there is no bins left anyway